static/app/components/events/viewHierarchy/tree.spec.tsx

```diff
@@ -0,0 +1,40 @@
+import {render, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import {ViewHierarchyTree} from './tree';
+
+const MOCK_DATA = {
+  id: 'parent',
+  type: 'Container',
+  children: [
+    {
+      id: 'intermediate',
+      type: 'Nested Container',
+      children: [
+        {
+          id: 'leaf',
+          type: 'Text',
+          children: [],
+        },
+      ],
+    },
+  ],
+};
+
+describe('View Hierarchy Tree', function () {
+  it('renders nested JSON', function () {
+    render(<ViewHierarchyTree hierarchy={MOCK_DATA} />);
+
+    expect(screen.getByText('Container')).toBeInTheDocument();
+    expect(screen.getByText('Nested Container')).toBeInTheDocument();
+    expect(screen.getByText('Text')).toBeInTheDocument();
+  });
+
+  it('can collapse and expand sections with children', function () {
+    render(<ViewHierarchyTree hierarchy={MOCK_DATA} />);
+
+    userEvent.click(screen.getAllByLabelText('Collapse')[1]);
+    expect(screen.queryByText('Text')).not.toBeInTheDocument();
+    userEvent.click(screen.getByLabelText('Expand'));
+    expect(screen.getByText('Text')).toBeInTheDocument();
+  });
```

Comment:
How can we make this whole thing more accessible?

Some ideas

* https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded
* Probably want to use `li` and `ul` for elements that build the tree
* Should the + and - be checkbox controls?

Reply:
Great points, thanks @evanpurkhiser

I took a look at the `aria-expanded` docs and added `aria-expanded` and `aria-controls` to the button to relate it to what it's toggling.

I agree that we should use `li` and `ul` here, so I've added that. I also used `details` and `summary` elements to be more semantic and handle the expand/collapse behaviour instead of rendering/not rendering children

> Should the + and - be checkbox controls?

I borrowed this button from the way we render JSON data on the issue details page. I can probably convert them to checkboxes but is that the correct element to be using for an expand/collapse control? I was looking for sources that back this up as a pattern and found some uses, but I also found that a button/button role might make more sense here

The mozilla docs mention "A button is a widget used to perform actions such as submitting a form, opening a dialog, canceling an action, or performing a command such as inserting a new record **or displaying information**" [[src](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#:~:text=or%20displaying%20information)] and the docs linked for `aria-expanded` use a button control. If we made it a checkbox would we have to set the `checked` property to `isExpanded`?

Reply:
Yeah I think you're right, button is probably good here. Thanks for making all these changes!!

cc @vuluongj20 wondering if we should generalize this component or not (also just want your takes on it)

This is awesome work Nar!!